test/metabase/test_runner.clj

```diff
@@ -0,0 +1,149 @@
+            eftest.runner
+            [environ.core :as env]
+            [metabase.config :as config]
+            [metabase.test-runner.init :as init]
+            [metabase.test-runner.junit :as junit]
+            [metabase.test-runner.parallel :as parallel]
+            [metabase.test.data.env :as tx.env]
+            metabase.test.redefs
+            [metabase.util :as u]
+            [pjstadig.humane-test-output :as humane-test-output]))
+
+;; initialize Humane Test Output if it's not already initialized.
+(humane-test-output/activate!)
+
+;; Load redefinitions of stuff like `tt/with-temp` and `with-redefs` that throw an Exception when they are used inside
+;; parallel tests.
+(comment metabase.test.redefs/keep-me)
+
+;;;; Finding tests
+
+(defmulti find-tests
+  "Find test vars in `arg`, which can be a string directory name, symbol naming a specific namespace or test, or a
+  collection of one or more of the above."
+  {:arglists '([arg])}
+  type)
+
+;; collection of one of the things below
+(defmethod find-tests clojure.lang.Sequential
+  [coll]
+  (mapcat find-tests coll))
+
+;; directory name
+(defmethod find-tests String
+  [dir-name]
+  (find-tests (io/file dir-name)))
+
+(def ^:private excluded-directories
+  "When searching the classpath for tests (i.e., if no `:only` options were passed), don't look for tests in any
+  directories with these name (as the last path component)."
+  #{"src"
+    "test_config"
+    "resources"
+    "test_resources"
+    "resources-ee"
+    "classes"})
+
+;; directory
+(defmethod find-tests java.io.File
+  [^java.io.File file]
+  (when (and (.isDirectory file)
+             (not (some (partial = (.getName file)) excluded-directories))
+             (if-let [[_ driver] (re-find #"modules/drivers/([^/]+)/" (str file))]
+               (contains? (tx.env/test-drivers) (keyword driver))
+               true))
+    (println "Looking for test namespaces in directory" (str file))
+    (->> (ns-find/find-namespaces-in-dir file)
+         (filter #(re-matches  #"^metabase.*test$" (name %)))
+         (mapcat find-tests))))
+
+;; a test namespace or individual test
+(defmethod find-tests clojure.lang.Symbol
+  [symb]
+  (if (namespace symb)
+    ;; a actual test var e.g. `metabase.whatever-test/my-test`
+    [symb]
+    ;; a namespace e.g. `metabase.whatever-test`
+    (binding [init/*test-namespace-being-loaded* symb]
+      (require symb)
+      (eftest.runner/find-tests symb))))
+
+;; default -- look in all dirs on the classpath
+(defmethod find-tests nil
+  [_]
+  (find-tests (classpath/system-classpath)))
+
+(defn tests [{:keys [only]}]
+  (when only
+    (println "Running tests in" (pr-str only)))
+  (let [tests (u/profile "Find tests" (find-tests only))]
+    (println "Running" (count tests) "tests")
+    tests))
+
+;;;; Running tests & reporting the output
+
+(defonce ^:private orig-test-var t/test-var)
+
+(defn run-test
+  "Run a single test `varr`. Wraps/replaces [[clojure.test/test-var]]."
```

Comment:
Nit: docstring has `varr`, but should it be `test-var`?

Reply:
Good catch